ehrapy/preprocessing/_balanced_sampling.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+if TYPE_CHECKING:
+    from anndata import AnnData
+
+from anndata import AnnData
+from imblearn.over_sampling import RandomOverSampler
+from imblearn.under_sampling import RandomUnderSampler
+
+
+def balanced_sample(
+    adata: AnnData,
+    key: str,
+    random_state: int = 0,
+    method: Literal["RandomUnderSampler", "RandomOverSampler"] = "RandomUnderSampler",
+    sampler_kwargs: dict = None,
+    copy: bool = False,
+) -> AnnData:
+    """Balancing groups in the dataset.
+
+    Balancing groups in the dataset based on group members in `.obs[key]` using the `imbalanced-learn <https://imbalanced-learn.org/stable/index.html>`_ package.
+    Currently supports `RandomUnderSampler` and `RandomOverSampler`.
+
+    Note that `RandomOverSampler <https://imbalanced-learn.org/stable/references/generated/imblearn.over_sampling.RandomOverSampler.html>`_ only replicates observations of the minority groups, which distorts several downstream analyses, very prominently neighborhood calculations and downstream analyses depending on that.
+    The `RandomUnderSampler <https://imbalanced-learn.org/stable/references/generated/imblearn.under_sampling.RandomUnderSampler.html>`_ by  default undersamples the majority group without replacement, not causing this issues of replicated observations.
+
+    Args:
+        adata: The annotated data matrix of shape `n_obs` × `n_vars`.
+        key: The key in `adata.obs` that contains the group information.
+        random_state: Random seed. Defaults to 0.
+        method: The method to use for balancing. Defaults to "RandomUnderSampler".
+        sampler_kwargs: Keyword arguments for the sample, see the `imbalanced-learn` documentation for options. Defaults to None.
```

Comment:
Not really important, but do you maybe mean _for the sampler_ or _for sampling_ here?

Reply:
ah, I see, thanks!

I meant for the sample_r_, I missed the r.